frontend/src/components/Layout/Header/Header.tsx

```diff
@@ -60,22 +67,32 @@ const Header: React.FC<HeaderProps> = ({
   handleConfirm,
   isMobileMenuOpen,
   toggleMobileMenu,
+  isSearchOpen,
+  searchCourses,
+  handleSearchClose,
+  coursesFound,
 }) => (
   <LayoutHeader container>
     <BrandLogoLink to={PATHS.profile}>
       <BrandLogo alt=":iTechArt" src={brandLogo} />
     </BrandLogoLink>
     <HeaderContent>
-      <Search
-        className="search"
-        disableUnderline
-        placeholder="Search"
-        startAdornment={
-          <InputAdornment position="start">
-            <SearchIcon color="disabled" fontSize="medium" />
-          </InputAdornment>
-        }
-      />
+      <ClickAwayListener onClickAway={handleSearchClose}>
+        <RelativeWrapper>
+          <Search
+            className="search"
+            disableUnderline
+            placeholder="Search"
+            startAdornment={
+              <InputAdornment position="start">
+                <SearchIcon color="disabled" fontSize="medium" />
+              </InputAdornment>
+            }
+            onChange={(event) => searchCourses(event.target.value)}
```

Comment:
We can replaced it to `onChange={searchCourses}`

Reply:
Do we really need to pass all the object to container, if we need only its value there?